packages/cli/README.md

```diff
@@ -1,210 +1,863 @@
+
+```sh
+npx tsx hello-world.ts --help # or -h
+```
+
+When you request help, the CLI will display information like this:
+
+```
+Hello World CLI
+
+Hello World CLI v1.0.0
+
+USAGE
+
+$ hello-world
+
+OPTIONS
+
+--completions sh | bash | fish | zsh
+
+  One of the following: sh, bash, fish, zsh
+
+  Generate a completion script for a specific shell
+
+  This setting is optional.
+
+(-h, --help)
+
+  A true or false value.
+
+  Show the help documentation for a command
+
+  This setting is optional.
+
+--wizard
+
+  A true or false value.
+
+  Start wizard mode for a command
+
+  This setting is optional.
+
+--version
+
+  A true or false value.
+
+  Show the version of the application
+
+  This setting is optional.
+```
+
+### Using the Wizard Mode
+
+The `--wizard` option activates the Wizard Mode in your CLI application, which provides a guided process for constructing commands. This is especially helpful for users who are new to your CLI or need assistance in building the correct command syntax.
+
+To initiate the Wizard Mode, run your CLI application with the `--wizard` option like this:
+
+```sh
+npx tsx hello-world.ts --wizard
+```
+
+When you start the Wizard Mode, the CLI will interactively guide you through the process of setting up a command. Here's what the interaction might look like:
+
+```
+Wizard Mode for CLI Application: Hello World CLI (v1.0.0)
+
+Instructions
+
+  The wizard mode will assist you with constructing commands for Hello World CLI (v1.0.0).
+
+  Please answer all prompts provided by the wizard.
+
+COMMAND: hello-world
+
+Wizard Mode Complete!
+
+You may now execute your command directly with the following options and arguments:
+
+    hello-world
+
+✔ Would you like to run the command? … yes / no
+```
+
+# Basic Usage
+
+## Adding Arguments to Commands
+
+Adding arguments to your commands allows your CLI applications to accept and process user input dynamically. Let's create a simple `echo` CLI that echoes back whatever text you pass to it.
+
+### Setting Up Your CLI
+
+Begin by creating a new TypeScript file named `echo.ts`. This file will contain all the code necessary to define a command that accepts user input as an argument.
+
+```ts
+// Import the necessary modules from the Effect libraries
+import { Args, Command } from "@effect/cli"
+import { NodeContext, NodeRuntime } from "@effect/platform-node"
+import { Console, Effect } from "effect"
+
+// Define a text argument
+const text = Args.text({ name: "text" })
+
+// Create a command that logs the provided text argument to the console
+const command = Command.make("echo", { text }, ({ text }) => Console.log(text))
+
+// Configure and initialize the CLI application
+const cli = Command.run(command, {
+  name: "Echo CLI",
+  version: "v0.0.1"
+})
+
+// Prepare and run the CLI application, providing necessary context and runtime
+Effect.suspend(() => cli(process.argv)).pipe(
```

Comment:
We can probably remove this `Effect.suspend` from all the samples since `process.argv` is statically defined on `process`. I should have probably done this in the example code.

Reply:
done